src/Ryujinx.Ui.Common/App/ApplicationLibrary.cs

```diff
@@ -506,26 +514,27 @@ public ApplicationMetadata LoadAndSaveMetaData(string titleId, Action<Applicatio
             string metadataFolder = Path.Combine(AppDataManager.GamesDirPath, titleId, "gui");
             string metadataFile   = Path.Combine(metadataFolder, "metadata.json");
 
-            ApplicationMetadata appMetadata;
+            bool isCached = _metadataCache.TryGetValue(metadataFile, out ApplicationMetadata appMetadata);
 
-            if (!File.Exists(metadataFile))
+            if (!isCached)
```

Comment:
```suggestion
            if (!_metadataCache.TryGetValue(metadataFile, out ApplicationMetadata appMetadata))
```

Since you just need the value of `isCached` once, I'd inline it.

Reply:
inlined. Also, it's reset on line 93 when a read from disk occurs.

There's no explicit cache file it's just reading the existing metadata entries as they're already written